### PR TITLE
Reset all WP Admin styles in the wrapper of the editor styles.

### PR DESCRIPTION
**Pull request by youknowriad**
Builds on top of #29588 

Extends that approach applied to headings to lists and paragraphs and also resets the "body" (editor-styles-wrapper) styles to the browser defaults. (There are still some styles on the editor-styles.scss files that I didn't check yet)

The idea now, is that by default if you use the "emptytheme", the frontend looks exactly like the backend by default and as you add styles to both your stylesheet (and loaded as part of the editor  styles), the frontend/backend parity will persist.

### CODE REVIEW

File: lib/client-assets.php [hunk 1]
```diff
@@ -664,6 +664,15 @@ function gutenberg_extend_block_editor_styles( $settings ) {
...
 	}
 
+	// Remove the default font editor styles.
+	// When Gutenberg is updated to have minimum version of WordPress 5.8
+	// This could be removed.
+	foreach ( $settings['styles'] as $j => $style ) {
+		if ( 0 === strpos( $style['css'], "body { font-family:" ) ) {
+			unset( $settings['styles'][ $j ] );
+		}
+	}
```
Comment (youknowriad) [inline lib/client-assets.php:674]: Core automatically loads a style setting a default font family, this removes that editor style (reset to browser defaults)

**Comment by scruffian** [comment]
This is looking really good. There are still some differences but I don't think they are caused by this.

One is the width of the editor:
Screenshot 2021-03-05 at 17 36 54

vs the width of the front end:
Screenshot 2021-03-05 at 17 37 02

The other one is the file block. Editor:
Screenshot 2021-03-05 at 17 35 58

Front end:
Screenshot 2021-03-05 at 17 35 58

The margins are also off but I think that will be done in a different issue?

**Comment by youknowriad** [comment]
yes, some of the margins (the ones relying on figure) are fixed here https://github.com/WordPress/gutenberg/pull/29517

And yeah, we won't be able to fix all in one go, we need to find a way to iterate and remove these differences bit by bit.

**Comment by jasmussen** [comment]
This is so cool.

**Comment by carolinan** [comment]
I think we need to do this but I am worried about how it will effect existing themes.
Especially the font family, because the browser defaults are really bad. 
Even with a dev note to alert theme authors, there are many themes that are not maintained and the users would be stuck with an editor that is not pleasant to work with.

**Comment by youknowriad** [comment]
Something I consider is whether we could load the "default font style" but only for themes that don't provide any custom editor styles. That was the whole original idea of that "editor-styles.scss" file but somehow/somewhere it changed it got loaded for all themes (and I didn't know about it)

**Review by carolinan — APPROVED** [review]

**Review by aristath — APPROVED** [review]
LGTM 👍

**Comment by gziolo** [comment]
I see that @carolinan has 5 merged commits and a lot of other types of contributions that is more than enough fro Gutenberg team member status 😄

**Comment by youknowriad** [comment]
@gziolo I thought it was already the case 😬 It should be fixed.

**Pull request merged**